onprc_ehr/src/org/labkey/onprc_ehr/issues/RestrictedIssueProviderImpl.java

```diff
@@ -0,0 +1,136 @@
+import org.labkey.api.issues.Issue;
+import org.labkey.api.issues.RestrictedIssueProvider;
+import org.labkey.api.query.SimpleValidationError;
+import org.labkey.api.query.ValidationError;
+import org.labkey.api.security.Group;
+import org.labkey.api.security.SecurityManager;
+import org.labkey.api.security.User;
+import org.labkey.api.security.ValidEmail;
+import org.labkey.api.util.Pair;
+
+import java.util.List;
+import java.util.Objects;
+
+public class RestrictedIssueProviderImpl implements RestrictedIssueProvider
+{
+    private static final String CAT_RESTRICTED_ISSUE_DEF_PROPERTIES = "RestrictedIssueDefProperties-";
+    private static final String PROP_RESTRICTED_ISSUE_LIST = "issueRestrictedIssueList";
+    private static final String PROP_RESTRICTED_ISSUE_LIST_GROUP = "issueRestrictedIssueListGroup";
+
+    @Override
+    public void setRestrictedIssueTracker(Container c, String issueDefName, Boolean isRestricted)
+    {
+        setPropertyValue(c, issueDefName, PROP_RESTRICTED_ISSUE_LIST, String.valueOf(isRestricted));
+    }
+
+    @Override
+    public boolean isRestrictedIssueTracker(Container c, String issueDefName)
+    {
+        String value = getPropertyValue(c, issueDefName, PROP_RESTRICTED_ISSUE_LIST);
+        if (value != null)
+        {
+            return Boolean.parseBoolean(value);
+        }
+        return false;
+    }
+
+    @Override
+    public void setRestrictedIssueListGroup(Container c, String issueDefName, @Nullable Group group)
+    {
+        setPropertyValue(c, issueDefName, PROP_RESTRICTED_ISSUE_LIST_GROUP, null != group ? String.valueOf(group.getUserId()) : "0");
+    }
+
+    @Override
+    public @Nullable Group getRestrictedIssueListGroup(Container c, String issueDefName)
+    {
+        String groupId = getPropertyValue(c, issueDefName, PROP_RESTRICTED_ISSUE_LIST_GROUP);
+        if (null == groupId)
+            return null;
+
+        return SecurityManager.getGroup(Integer.valueOf(groupId));
+    }
+
+    private void setPropertyValue(Container c, String issueDefName, String key, String value)
+    {
+        PropertyManager.PropertyMap props = PropertyManager.getWritableProperties(c, getPropMapName(issueDefName), true);
+        props.put(key, value);
+        props.save();
+    }
+
+    private String getPropertyValue(Container c, String issueDefName, String key)
+    {
+        return PropertyManager.getProperties(c, getPropMapName(issueDefName)).get(key);
+    }
+
+    private String getPropMapName(String issueDefName)
+    {
+        if (issueDefName == null)
+            throw new IllegalArgumentException("Issue def name must be specified");
+
+        return CAT_RESTRICTED_ISSUE_DEF_PROPERTIES + issueDefName;
+    }
+
+    @Override
+    public boolean hasPermission(User user, @NotNull Issue issue, List<Issue> relatedIssues, List<ValidationError> errors)
+    {
+        Container issueContainer = ContainerManager.getForId(issue.getContainerId());
```

Comment:
if the `issueContainer` is null, it looks like we'll fall through to the bottom and return `true`. is that expected? I assume we would probably hit issues before this method if that `issueContainer` was null.

Reply:
yes, this becomes a `noop` and most likely will fail in whatever action the issue was requested for.